src/checkers/inference/InferenceLauncher.java

```diff
@@ -0,0 +1,355 @@
+            String current = argList.get(i);
+            if (current.startsWith("-Xmx") || current.startsWith("-Xms")) {
+                argList.remove(i);
+            } else {
+                ++i;
+            }
+        }
+    }
+
+    public List<String> insertJaif() {
+        List<String> outputJavaFiles = new ArrayList<>(InferenceOptions.javaFiles.length);
+
+        printStep("Inserting annotations", outStream);
+        int result;
+        if (!InferenceOptions.inPlace) {
+            final File outputDir = new File(InferenceOptions.afuOutputDir);
+            TestUtilities.ensureDirectoryExists(outputDir);
+
+            String jaifFile = getJaifFilePath(outputDir);
+
+            String [] options = new String [5 + InferenceOptions.javaFiles.length];
+            options[0] = "insert-annotations-to-source";
+            options[1] = "-v";
+            options[2] = "-d";
+            options[3] = outputDir.getAbsolutePath();
+            options[4] = jaifFile;
+
+            System.arraycopy(InferenceOptions.javaFiles, 0, options, 5, InferenceOptions.javaFiles.length);
+
+            if (InferenceOptions.printCommands) {
+                outStream.println("Running Insert Annotations Command:");
+                outStream.println(PluginUtil.join(" ", options));
+            }
+
+            //this can get quite large for large projects and it is not advisable to run
+            //roundtripping via the InferenceLauncher for these projects
+            ByteArrayOutputStream insertOut = new ByteArrayOutputStream();
+            result = ExecUtil.execute(options, insertOut, errStream);
+            outStream.println(insertOut.toString());
+
+
+            List<File> newJavaFiles = findWrittenFiles(insertOut.toString());
+            for (File newJavaFile : newJavaFiles) {
+                outputJavaFiles.add(newJavaFile.getAbsolutePath());
+            }
+
+        } else {
+            String jaifFile = getJaifFilePath(new File("."));
+
+            String [] options = new String [5 + InferenceOptions.javaFiles.length];
+            options[0] = "insert-annotations-to-source";
+            options[1] = "-v";
+            options[2] = "-i";
+            options[4] = jaifFile;
+
+            System.arraycopy(InferenceOptions.javaFiles, 0, options, 5, InferenceOptions.javaFiles.length);
+
+            if (InferenceOptions.printCommands) {
+                outStream.println("Running Insert Annotations Command:");
+                outStream.println(PluginUtil.join(" ", options));
+            }
+
+            result = ExecUtil.execute(options, outStream, errStream);
+
+            for (String filePath : InferenceOptions.javaFiles) {
+                outputJavaFiles.add(filePath);
+            }
+        }
+
+        reportStatus("Insert annotations", result, outStream);
+        outStream.flush();
+        exitOnNonZeroStatus(result);
+        return outputJavaFiles;
+    }
+
+    private static List<File> findWrittenFiles(String output) {
```

Comment:
This method interprets the output of the AFU and determines where the updated java files have been written.